### ISSUE TITLE
Typescript definitions not generated correctly

### ISSUE THREAD

**Issue by liviuignat** [Codeception/CodeceptJS#1012]
#### What are you trying to achieve?

Generate typescript definitions

#### What do you get instead?

- Typescript definition file is not generated correctly. Please see the image bellow, where multiple functions definitions are not correct:

![whatsapp image 2018-04-01 at 16 06 12](https://user-images.githubusercontent.com/2430288/38174064-ae753bc8-35c7-11e8-85ec-5285bba85a9a.jpeg)

#### Sugestions:

- add tests to see that definitions are generated correctly
- having `.d.ts` files in your project directly are not the preffered way anymore, rather having a module `@types/codeceptjs`
- please add more types to functions, most of them use `any`

Command used: `codeceptjs def .`

### Details

* CodeceptJS version: 1.1.7
* NodeJS Version: 8.11.0
* Operating System: MacOS
* WebDriverIO
* Configuration file: 
```json
{
  "bootstrap": "server.typescript.js",
  "tests": "src/*/.test.ts",
  "timeout": 10000,
  "output": "./output",
  "helpers": {
    "WebDriverIO": {
      "url": "http://localhost",
      "browser": "chrome",
      "restart": true
    }
  },
  "include": {
    "I": "./steps_file.js"
  },
  "mocha": {},
  "name": "codecept"
}
```

**Comment by ontytoom** [comment]
Defect confirmed on:
* CodeceptJS 1.1.7
* OS: Windows 7

I have to manually edit the generated `steps.d.ts` in order to make it parse correctly.

**Comment by timbergfreund** [comment]
same issue at
- CodeceptJS 1.1.7
- OS: Ubunutu 16.04 LTS

**Comment by DavertMik** [comment]
Yes, TS support should be improved, for sure.

**Comment by DavertMik** [comment]
@liviuignat 
> having .d.ts files in your project directly are not the preffered way anymore, rather having a module @types/codeceptjs

Definitions are generated depending on a config so it can't be moved out to another repo

**Comment by liviuignat** [comment]
@DavertMik I know.

Doing thins in `package.json` should also be enough:

```json
  "name": "codeceptjs",
  "version": "1.1.10"
  "main": "lib/index.js",
  "types": "typings/index.d.ts",
```

**Comment by DavertMik** [comment]
yeah, but still we don't have much of internal types to be used. We generate types for `I` object which consists of methods taken from helpers.

Anyway, the TS generator was broken in 1.1.7 but it should be fixed in 1.1.8. 

So now by running `codeceptjs def` you get a list of commands with types:

```ts
declare namespace CodeceptJS {
  export interface I {
    amOutsideAngularApp() : void,
    amInsideAngularApp() : void,
    amOnPage(url: string) : void,
    click(locator: ILocator, context?: ILocator) : void,
    click(locator: string, context?: ILocator) : void,
    click(locator: ILocator, context?: string) : void,
    click(locator: string, context?: string) : void,
    doubleClick(locator: ILocator, context?: ILocator) : void,
    doubleClick(locator: string, context?: ILocator) : void,
    doubleClick(locator: ILocator, context?: string) : void,
    doubleClick(locator: string, context?: string) : void,
    rightClick(locator: ILocator, context?: ILocator) : void,
    rightClick(locator: string, context?: ILocator) : void,
    rightClick(locator: ILocator, context?: string) : void,
    rightClick(locator: string, context?: string) : void,
    moveCursorTo(locator: ILocator, offsetX?: string, offsetY?: string) : void,
    moveCursorTo(locator: string, offsetX?: string, offsetY?: string) : void,
    see(text: string, context?: ILocator) : void,
    see(text: string, context?: string) : void,
    seeTextEquals(text: string, context?: ILocator) : void,
    seeTextEquals(text: string, context?: string) : void,
    dontSee(text: string, context?: ILocator) : void,
    dontSee(text: string, context?: string) : void,
    grabBrowserLogs() : Promise,
    grabCurrentUrl() : Promise,
    selectOption(select: ILocator, option: string) : void,
    selectOption(select: string, option: string) : void,
    fillField(field: ILocator, value: string) : void,
    fillField(field: string, value: string) : void,

```
